packages/fx-core/src/core/middleware/projectMigrator.ts

```diff
@@ -409,6 +412,40 @@ async function removeBotConfig(ctx: CoreHookContext) {
     delete envConfig[ResourcePlugins.Bot];
     envConfig[ResourcePlugins.Bot] = { wayToRegisterBot: "create-new" };
   }
+  let needUpdate = false;
+  let configPrefix = "";
+  if (envConfig["solution"]) {
+    if (envConfig["solution"]["subscriptionId"] && envConfig["solution"]["resourceGroupName"]) {
+      configPrefix = `/subscriptions/${envConfig["solution"]["subscriptionId"]}/resourcegroups/${envConfig["solution"]["resourceGroupName"]}`;
+      needUpdate = true;
+    }
+  }
+  if (needUpdate && envConfig[ResourcePlugins.FrontendHosting]) {
+    envConfig[ResourcePlugins.FrontendHosting][
+      EnvConfigName.StorageResourceId
+    ] = `${configPrefix}/providers/Microsoft.Storage/storageAccounts/${
+      envConfig[ResourcePlugins.FrontendHosting][EnvConfigName.StorageName]
+    }`;
+  }
+  if (needUpdate && envConfig[ResourcePlugins.AzureSQL]) {
+    envConfig[ResourcePlugins.AzureSQL][
+      EnvConfigName.SqlResourceId
+    ] = `${configPrefix}/providers/Microsoft.Sql/servers/${
+      envConfig[ResourcePlugins.AzureSQL][EnvConfigName.SqlEndpoint].split(
```

Comment:
The sql endpoint will exist after provision. Will the migration happen in scaffolding step?

Reply:
no, this middle ware not trigger on scaffolding step